File: backend/src/utils/whatsappMessage.js
function buildWhatsappMessage({ orderCode, results, baseUrl }) {
    // ✅ Limpiar baseUrl para evitar doble slash (//)
    const cleanBaseUrl = String(baseUrl || "").replace(/\/+$/, "");

    const lines = [];
    lines.push(`🧾 Orden: ${orderCode}`);
    lines.push(`📦 Pedido múltiple (${results.length} items)`);
    lines.push("");

    for (const r of results) {
        const yyyy = r.expiresAt.toISOString().slice(0, 10);
        const credentialUrl = `${cleanBaseUrl}/s/${r.token}`;

        const template = r?.plan?.whatsapp_template;

        if (template && String(template).trim() !== "") {
            // Flujo Opción 2: Plantilla absoluta
            const pinVal = r?.account?.pin || "";
            const profVal = r?.account?.profile_number || "";
            const emailVal = r?.account?.email || "";
            const passVal = r?.account?.password || "";
            const platName = r?.plan?.platform_name || "";

            let finalMsg = String(template)
                .replace(/{PLATAFORMA}/gi, platName)
                .replace(/{CORREO}/gi, emailVal)
                .replace(/{PASSWORD}/gi, passVal)
                .replace(/{CLAVE}/gi, passVal) // alias
                .replace(/{PIN}/gi, pinVal)
                .replace(/{PERFIL}/gi, profVal)
                .replace(/{EXPIRA}/gi, yyyy)
                .replace(/{URL}/gi, credentialUrl)
                .replace(/{ENLACE}/gi, credentialUrl); // alias

            lines.push(finalMsg);
            lines.push("");

        } else {
            // Flujo original (Fallback)
            lines.push(`🆔 ID: ${r.subscriptionId} | 🖥️ ${r.plan.platform_name}`);
            lines.push(`📧 Correo: ${r.account.email}`);
            lines.push(`🔑 Contraseña: ${r.account.password}`);

            const profile = r?.account?.profile_number;
            if (profile !== null && profile !== undefined && String(profile).trim() !== "") {
                lines.push(`👤 Perfil: ${profile}`);
            }

            const pin = r?.account?.pin;
            if (pin !== null && pin !== undefined && String(pin).trim() !== "") {
                lines.push(`🔢 Pin: ${pin}`);
            }

            lines.push(`📅 Expira: ${yyyy}`);
            lines.push("");
            lines.push("");
        }
    }

    return lines.join("\n").trim();
}

module.exports = { buildWhatsappMessage };

File: go-backend/store-service/utils/whatsappMessage.go
package utils

import (
	"fmt"
	"strings"
	"time"

	"pagv2strbx-store/models"
)

type WhatsappResultItem struct {
	SubscriptionID int
	PlanName       string
	Account        models.PlatformAccount
	ExpiresAt      time.Time
	Token          string
}

func BuildWhatsappMessage(orderCode string, results []WhatsappResultItem, baseUrl string) string {
	cleanBaseUrl := strings.TrimRight(baseUrl, "/")

	var lines []string
	lines = append(lines, fmt.Sprintf("🧾 Orden: %s", orderCode))
	lines = append(lines, fmt.Sprintf("📦 Pedido múltiple (%d items)", len(results)))
	lines = append(lines, "")

	for _, r := range results {
		yyyy := r.ExpiresAt.Format("2006-01-02")
		credentialUrl := fmt.Sprintf("%s/s/%s", cleanBaseUrl, r.Token)

		lines = append(lines, fmt.Sprintf("🆔 ID: %d | 🖥️ %s", r.SubscriptionID, r.PlanName))
		lines = append(lines, fmt.Sprintf("📧 Correo: %s", r.Account.Email))
		lines = append(lines, fmt.Sprintf("🔑 Contraseña: %s", r.Account.Password))

		if r.Account.ProfileNumber > 0 {
			lines = append(lines, fmt.Sprintf("👤 Perfil: %d", r.Account.ProfileNumber))
		}

		if strings.TrimSpace(r.Account.Pin) != "" {
			lines = append(lines, fmt.Sprintf("🔢 Pin: %s", r.Account.Pin))
		}

		lines = append(lines, fmt.Sprintf("📅 Expira: %s", yyyy))
		lines = append(lines, fmt.Sprintf("*🔗⚠️ Debido a que en ocasiones se bloquea o cambia la clave, en este enlace %s puedes consultar la contraseña hasta tu último día contratado. 💻🔑:*", credentialUrl))
		lines = append(lines, "")
		lines = append(lines, "")
	}

	return strings.Join(lines, "\n")
}
